mesh/mesh.cpp

```diff
@@ -6596,6 +6596,7 @@ int *Mesh::GeneratePartitioning(int nparts, int part_method)
          for (i = 0; i < nparts; i++)
```

Comment:
I think this code (lines 6596-611) should be factored out into a function (or even as a lambda function defined within `Mesh::GeneratePartitioning`) since it is identical to lines 6544-6559. It looks like this was a copy-paste bug that could have been avoided.

Reply:
e.g.
```c++
auto compute_partition_sizes = [&]()
{
   for (i = 0; i < nparts; i++)
   {
      psize[i].one = 0;
      psize[i].two = i;
   }

   for (i = 0; i < NumOfElements; i++)
   {
      psize[partitioning[i]].one++;
   }

   empty_parts = 0;
   for (i = 0; i < nparts; i++)
   {
      if (psize[i].one == 0) { empty_parts++; }
   }
};
```

Reply:
Ok, I just made this change and added a short comment describing `psize`.

Reply:
Thanks!